src/Microsoft.AspNet.Mvc.Core/NonControllerAttribute.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Indicates that the type and any derived types that this attribute is applied to
+    /// is not considered a controller by the default controller discovery mechanism.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
```

Comment:
Kinda odd to inherit this thing. By inheriting it you're essentially not allowing people to derive from types which aren't controllers.  Great example:

```C#
// I want this to be my custom base (with existing Controller helper methods), but don't want it to be an exposed controller.
[NotController]
public class CustomController : Controller {}

// Didn't want my custom base class to be a controller but I DO want this one to be.
public class MyFrameworkController : CustomController {}
```